### PR TITLE
Feature/frontend filter

### PR DESCRIPTION
**Pull request by Mikazian**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [ ] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [x] I requested code review from other team members

## Frontend checklist:

- [x] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [x] My `Javascript` generate no new console errors
- [x] I tested my code cross browsers
- [x] My slice is pixel perfect for both desktop and mobile according to design
- [x] I conducted basic QA to assure all features are working
- [x] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Comment by Mikazian** [comment]
J'ai refais un repull une autre branche pour faire un format prettier de mon code car ça marchait pas sur l'autre branche. J'ai close l'autre branche. Bun run format a enlevé les virgules des fichier ts.config et les modifications demandées ont étés apportés :
- correction : ajouté le cleanup dans use-languages.ts et enlevé la classe vide dans FilterLanguagesForm.tsx

**Review by greegouze — DISMISSED** [review]

File: packages/frontend/web/src/contexts/PreferenceContext.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  useEffect(() => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+    fetchPreferences({ signal }).catch(() => {
+      setErrorPreferences(
+        'ⓘ An error occurred while fetching user preferences.',
+      );
+    });
+  }, [fetchPreferences]);
```
Comment (YDrogen) [inline packages/frontend/web/src/contexts/PreferenceContext.tsx:101]: cleanup function manquante

File: packages/frontend/web/src/hooks/use-languages.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+      setLanguages(data);
+    };
+
+    fetchLanguages().catch(() => {
+      setErrorLanguages('Fail to fetch languages');
+
+      return () => {
+        controller.abort();
+      };
```
Comment (YDrogen) [inline packages/frontend/web/src/hooks/use-languages.ts:29]: À déplacer hors du catch

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/contexts/PreferenceContext.tsx (hunk 1 above)
Comment (Mikazian) [reply packages/frontend/web/src/contexts/PreferenceContext.tsx:101]: corrigé

File: packages/frontend/web/src/hooks/use-languages.ts (hunk 2 above)
Comment (Mikazian) [reply packages/frontend/web/src/hooks/use-languages.ts:29]: corrigé, j'avais un probleme avec la fonction cleanup j'ai fais une fonction anonyme

**Review by greegouze — APPROVED** [review]

File: packages/frontend/web/src/contexts/PreferenceContext.tsx (hunk 1 above)
Comment (YDrogen) [reply packages/frontend/web/src/contexts/PreferenceContext.tsx:101]: Non, il manque toujours la cleanup function 🤔

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/contexts/PreferenceContext.tsx (hunk 1 above)
Comment (Mikazian) [reply packages/frontend/web/src/contexts/PreferenceContext.tsx]: c'est bon il a ajouté

**Review by greegouze — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**